pkg/services/datasources/service/datasource_service.go

```diff
@@ -128,7 +136,23 @@ func (s *Service) AddDataSource(ctx context.Context, cmd *models.AddDataSourceCo
 		return err
 	}
 
-	return s.SQLStore.AddDataSource(ctx, cmd)
+	if err := s.SQLStore.AddDataSource(ctx, cmd); err != nil {
+		return err
+	}
+
+	if s.features.IsEnabled(featuremgmt.FlagAccesscontrol) {
+		if _, err := s.permissionsService.SetPermissions(ctx, cmd.OrgId, strconv.FormatInt(cmd.Result.Id, 10), accesscontrol.SetResourcePermissionCommand{
```

Comment:
Didn't think of this during our call, but do you think we should leverage the hook system you implemented to perform the `AddDataSource` in the same session ? It would mean modifying the empty service, though.

Reply:
Nevermind, discussed with you, we wouldn't want to call the hook to add a data-source every time we set a permission!